addons/mail/models/discuss/discuss_channel.py

```diff
@@ -607,13 +607,19 @@ def _notify_by_web_push_prepare_payload(self, message, msg_vals=False):
         payload = super()._notify_by_web_push_prepare_payload(message, msg_vals=msg_vals)
         payload['options']['data']['action'] = 'mail.action_discuss'
         record_name = msg_vals.get('record_name') if msg_vals and 'record_name' in msg_vals else message.record_name
+        author_id = [msg_vals["author_id"]] if msg_vals.get("author_id") else message.author_id.ids
```

Comment:
Note that this was already incorrect [here](https://github.com/odoo/odoo/pull/187713/files#diff-ec9b6d3d5a022b3d96672121c5850b4a4b876c2f8f4dd0ac6f5b9266d310eaccL1350). `msg_vals` can be `False`, in which case this line will raise an `AttributeError`. Before this PR, it would have been a `TypeError`. Nevertheless, it’s better to fix it.

Reply:
Don't think it's practically possible to have a crash in 17.0: this is necessarily set: https://github.com/odoo/odoo/blob/17.0/addons/mail/models/mail_thread.py#L4422

Reply:
Nice to know. However, since the method allows this parameter to be omitted, it’s a bit concerning not to handle cases where the parameter is indeed omitted.